pkg/monitor/aws.go

```diff
@@ -0,0 +1,160 @@
+package monitor
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/elb"
+)
+
+type AWSLoadBalancerConfig struct {
+	Name      string
+	Instances []string
+}
+
+type AWSManager interface {
+	Instances(tags map[string]string) ([]string, error)
+	SyncLoadBalancer(cfg AWSLoadBalancerConfig) error
+}
+
+func newAWSManager(region string) AWSManager {
+	awscfg := aws.NewConfig()
+	awscfg = awscfg.WithRegion(region)
+	awscfg = awscfg.WithCredentialsChainVerboseErrors(true)
+
+	return &awsManager{
+		cfg: awscfg,
+	}
+}
+
+type awsManager struct {
+	cfg *aws.Config
+}
+
+func (am *awsManager) Instances(tags map[string]string) ([]string, error) {
+	svc := ec2.New(session.New(am.cfg))
+
+	var filters []*ec2.Filter
+	for k, v := range tags {
+		filters = append(filters, &ec2.Filter{
+			Name:   aws.String(fmt.Sprintf("tag:%s", k)),
+			Values: []*string{aws.String(v)},
+		})
+	}
+
+	req := ec2.DescribeInstancesInput{
+		Filters: filters,
+	}
+
+	resp, err := svc.DescribeInstances(&req)
+	if err != nil {
+		return nil, err
+	}
+
+	instances := []string{}
+
+	//TODO(bcwaldon): need to handle resp.NextToken
```

Comment:
I'm intuiting this to occur when the number of `Reservations` is larger than the page size. How many is that?